the prepend command

aryarm opened the pull request:
Note: Make sure you decide whether to merge #88 before you merge this one.

Ever since I realized that I could portably prepend text to a file using the syntax described in issue #72, I've been using the idiom quite often. Unfortunately, it's quite lengthy and sometimes hard to remember, so I decided to whip up a quick function alias for it.

## Motivation
Prepending text to a file can be a difficult task. The classic idiom is to use GNU `sed`'s in-place editing feature `-i`. However, `sed -i` (as well as most other methods) use a temporary file in the background. In addition, not every user will have GNU `sed` installed, so this strategy isn't portable.

The `prepend` command is a portable way to prepend text to a file without creating a temporary file. This is usually faster because it allows one to bypass unnecessary file IO operations. Text can be specified either from stdin or as arguments to the command. In addition, `prepend` is flexible enough to work with almost any type of file, including named pipes. Note, however, that `prepend` is not recommended for large files.

## Usage
```
prepend <file> <text>
cmd | prepend <file>
```

aryarm committed:
create new prepend command (based on discussion in issue #72)

aryam7 commented:
## An alternative implementation
(adapted from https://stackoverflow.com/a/14077361)
```
cmd | { echo '0a' && cat && echo -e ".\nw"; } | ed file 2>/dev/null
```
I don't know how this works but it does. Supposedly, `ed` creates an in memory buffer before it writes to the file. `ed` seems to be POSIX so I think it's technically portable. And it might even be better than my current solution?

aryarm committed:
allow setup to prepend to bash_aliases instead. also do a refactor

aryarm committed:
use prepend in setup

aryarm committed:
declare ABSPATH in .bash_aliases instead of .included so that it's easy to change ABSPATH programmatically

aryarm committed:
Allow src command to handle cases where the default shell is installed locally
Sometimes, the user won't have sudo access and their default shell might be installed
in their $HOME directory instead of in /bin

aryarm committed:
fix some quoting errors after testing

aryarm committed:
add comment for src cmd

aryarm committed:
update uninstall to support new setup

aryarm committed:
handle cases where param1 might be invalid or empty. also remove unnecessary check for file existence

aryarm committed:
Merge branch 'prepend' of github.com:GiselleSerate/myaliases into prepend

GiselleSerate committed:
Merge pull request #88 from GiselleSerate/new-setup

new setup script
Looks good. Let's do it.

GiselleSerate commented:
I'm gonna say we should catch usage errors (e.g. no arguments) for prepend

GiselleSerate committed:
Add instructional sourcing print statement

aryam7 commented:
good point!
see 63d2236

GiselleSerate commented:
Wait, it looks like you tried to make it error out or something--when I try to run `prepend` on its own with no arguments I get:
```
touch: cannot touch '': No such file or directory 

```
And then it hangs, presumably waiting for input, until I Ctrl-C out.

aryam7 commented:
That was the old behavior. Are you sure you're working with the new code?
You can do `type prepend` to check.

GiselleSerate commented:
Oh boy--maybe I merged badly. Lemme look.
